### DISCUSSION TITLE
Deprecate rowOverEffect/rowOutEffect and use nth-child(odd/even) for row highlighting

### DISCUSSION THREAD

**Discussion by neekfenwick** [zencart/zencart#6470] [Ideas]
Not sure if this has been covered before but it has been bugging me for years.  for 10 or 15 years, CSS has provided `:hover` and `:nth-child(odd/even)` to achieve what the Javascript event handlers `rowOverEffect()` etc do.

Is there general approval of deprecating these?

I imagine we need to keep the functions and class definitions around, for legacy addons that use them, but I'd be happy to modify core ZC to stop using these ancient techniques.

From a quick search, maybe the odd/even styling isn't actually used much.. I see mention of it here but it doesn't seem to be used in the associated PHP code.  Is `responsive_classic` deprecated and can be ignored?
https://github.com/zencart/zencart/blob/e3e94348e20e7075f8a728bfb4623129c145543f/includes/templates/responsive_classic/css/responsive_default.css#L117

**Comment by drbyte** [comment]
I agree: the old odd/even and `rowOverEffect()` stuff can safely be replaced with CSS. 
Even the Bootstrap framework already uses CSS things for it, so makes it easy to apply by adding the right utility class to the markup.

There are some mouseover things too, I think mostly in the zen_image() function (and maybe others in html_output.php too) that would be great to rip out .... but they're quite fragile. That's another topic. 

I'm not sure what you meant by "Is `responsive_classic` deprecated and can be ignored?" ... because `responsive_classic` is the first enabled template a storeowner currently sees when installing Zen Cart.

**Reply by neekfenwick** [reply]
Thanks, yes, I wasn't sure which template is the default.  When I run into something as ancient as PHP variables to assign even/odd classes, or a javascript rowover handler to swap CSS classes, it certainly smells like long dead code :)  As usual, if I find some time to attack this rowover stuff I'll consider making a PR, but if anyone else beats me to it maybe this thread would be a place to announce it so we don't duplicate effort.
